File: java/src/Person.java
public class Person {
	private String name;
	private int byear;

	public Person(String name, int byear){
		this.name = name;
		this.byear = byear;
	}

	public void info(){
		System.out.println("My name is: " + name);
		System.out.println("My birth year is: " + byear);
	}

	public static void main(String[] args) {
		Person p = new Person("Karim", 1986);
		p.info();
	}
}

File: perl/person.pl
#! /usr/bin/perl -w

{
	package Person;

	sub new {
		my $class = shift;
		my $self = {
			_name => shift,
			_byear => shift,
		};
		bless $self, $class;
		return $self;
	}

	sub info {
		my( $self ) = @_;
		print "My name: $self->{_name}\n";
		print "My birth year: $self->{_byear}\n";
	}
}

$p = new Person("Karim", 1986);

$p->info();
